build-logic/project-plugins/src/main/kotlin/com.hedera.hashgraph.jpms-modules.gradle.kts

```diff
@@ -14,284 +14,209 @@
-        exportAllPackages()
-        requires("org.connid.framework") // this is missing in POM
+    withModule<RemoveDependenciesMetadataRule>("com.google.protobuf:protobuf-java-util") {
+        params(annotationLibraries)
     }
-    module("org.jetbrains:annotations", "org.jetbrains.annotations") {
-        exportAllPackages()
-        // no dependencies
+    withModule<RemoveDependenciesMetadataRule>("io.helidon.grpc:io.grpc") {
+        params(annotationLibraries)
     }
-    module("io.tmio:tuweni-units", "tuweni.units") {
-        exportAllPackages()
-        requireAllDefinedDependencies()
+    withModule<RemoveDependenciesMetadataRule>("org.apache.tuweni:tuweni-bytes") {
+        params(annotationLibraries)
     }
-    module("io.tmio:tuweni-bytes", "tuweni.bytes") {
-        exportAllPackages()
-        requireAllDefinedDependencies()
+    withModule<RemoveDependenciesMetadataRule>("org.apache.tuweni:tuweni-units") {
+        params(annotationLibraries)
     }
-    module("net.i2p.crypto:eddsa", "net.i2p.crypto.eddsa") {
-        exportAllPackages()
-        // no dependencies
+    withModule<RemoveDependenciesMetadataRule>("io.prometheus:simpleclient") {
+        params(
+            listOf(
+                "io.prometheus:simpleclient_tracer_otel",
+                "io.prometheus:simpleclient_tracer_otel_agent"
+            )
+        )
     }
-    module("io.netty:netty-codec-http", "io.netty.codec.http") {
-        exportAllPackages()
-        requireAllDefinedDependencies()
+    withModule<RemoveDependenciesMetadataRule>("org.jetbrains.kotlin:kotlin-stdlib") {
+        params(listOf("org.jetbrains.kotlin:kotlin-stdlib-common"))
     }
-    module("io.netty:netty-codec-http2", "io.netty.codec.http2") {
-        exportAllPackages()
-        requireAllDefinedDependencies()
+    withModule<RemoveDependenciesMetadataRule>("junit:junit") {
+        params(listOf("org.hamcrest:hamcrest-core"))
     }
-    module("io.netty:netty-codec-socks", "io.netty.codec.socks") {
+}
+
+// Fix or enhance the 'module-info.class' of third-party Modules. This is about the
+// 'module-info.class' inside the Jar files. In our full Java Modules setup every
+// Jar needs to have this file. If it is missing, it is added by what is configured here.
+extraJavaModuleInfo {
+    failOnAutomaticModules.set(true) // Only allow Jars with 'module-info' on all module paths
+
+    module("io.grpc:grpc-netty", "grpc.netty") {
         exportAllPackages()
         requireAllDefinedDependencies()
+        requires("java.logging")
     }
-    module("io.netty:netty-handler-proxy", "io.netty.handler.proxy") {
+    module("io.grpc:grpc-stub", "grpc.stub") {
         exportAllPackages()
         requireAllDefinedDependencies()
+        requires("java.logging")
     }
-    module("io.netty:netty-transport-native-unix-common", "io.netty.transport.unix.common") {
+    module("io.grpc:grpc-testing", "grpc.testing")
```

Comment:
We can shirnk this down now as 

```
exportAllPackages()
requireAllDefinedDependencies()
```

Is the default convention in the latest release of the plugin - https://github.com/gradlex-org/extra-java-module-info/issues/99